arangod/Aql/OptimizerUtils.cpp

```diff
@@ -688,6 +691,28 @@ std::pair<bool, bool> getBestIndexHandlesForFilterCondition(
 
   TRI_ASSERT(usedIndexes.empty());
 
+  // we might have an inverted index - it could cover whole condition at once.
+  // Give it a try
+  for (auto& index : indexes) {
+    if (index.get()->type() == Index::TRI_IDX_TYPE_INVERTED_INDEX &&
+        ((hint.type() == IndexHint::Simple && // apply this index only if the hint-а
+          std::find(hint.hint().begin(), hint.hint().end(),
+                    index->name()) != hint.hint().end()))) {
+      auto costs = index.get()->supportsFilterCondition(indexes, root, reference, itemsInCollection);
+      if (costs.supportsCondition) {
+        // we need to find 'root' in 'ast' and replace it with specialized version
+        // but for now we know that index will not alter the node, so just an assert
+        index.get()->specializeCondition(root, reference);
+        usedIndexes.emplace_back(index);
+        isAllCoveredByIndex = true;
+        // FIXME: we should somehow consider other indices and calculate here "overall" score
+        // Also a question: if sort is covered but filter is not ? What is more optimal?
+        auto const sortSupport = index.get()->supportsSortCondition(sortCondition, reference, itemsInCollection);
+        return std::make_pair(true, sortSupport.supportsCondition);
```

Comment:
It looks like we would always prefer inverted index now. Not saying this is wrong, but if there is a matching inverted index, it will always be preferred over other index types (and even over other index types that support both the filter and the sort condition).

Reply:
No, inverted index for now is only permitted by explicit hint.